src/_ert/forward_model_runner/client.py

```diff
@@ -26,107 +22,139 @@ class ClientConnectionClosedOK(Exception):
-        if max_retries is None:
-            max_retries = self.DEFAULT_MAX_RETRIES
-        if timeout_multiplier is None:
-            timeout_multiplier = self.DEFAULT_TIMEOUT_MULTIPLIER
-        if url is None:
-            raise ValueError("url was None")
+        self._ack_timeout = ack_timeout or self.DEFAULT_ACK_TIMEOUT
         self.url = url
         self.token = token
-        self._additional_headers = Headers()
+
+        # Set up ZeroMQ context and socke
+        self._ack_event: asyncio.Event = asyncio.Event()
+        self.context = zmq.asyncio.Context()
+        self.socket = self.context.socket(zmq.DEALER)
+        self.socket.setsockopt(zmq.LINGER, 0)
+        if dealer_name is None:
+            self.dealer_id = f"dispatch-{uuid.uuid4().hex[:8]}"
+        else:
+            self.dealer_id = dealer_name
+        self.socket.setsockopt_string(zmq.IDENTITY, self.dealer_id)
+        print(f"Created: {self.dealer_id=} {token=} {self._ack_timeout=}")
         if token is not None:
-            self._additional_headers["token"] = token
-
-        # Mimics the behavior of the ssl argument when connection to
-        # websockets. If none is specified it will deduce based on the url,
-        # if True it will enforce TLS, and if you want to use self signed
-        # certificates you need to pass an ssl_context with the certificate
-        # loaded.
-        self._ssl_context: bool | ssl.SSLContext | None = None
-        if cert is not None:
-            self._ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
-            self._ssl_context.load_verify_locations(cadata=cert)
-        elif url.startswith("wss"):
-            self._ssl_context = True
-
-        self._max_retries = max_retries
-        self._timeout_multiplier = timeout_multiplier
-        self.websocket: ClientConnection | None = None
+            client_public, client_secret = zmq.curve_keypair()
+            self.socket.curve_secretkey = client_secret
+            self.socket.curve_publickey = client_public
+            self.socket.curve_serverkey = token.encode("utf-8")
+
         self.loop = new_event_loop()
+        self._receiver_task = None
+
+    async def connect(self) -> None:
+        self.socket.connect(self.url)
+        await self._term_receiver_task()
+        self._receiver_task = asyncio.create_task(self._receiver())
+        try:
+            await self._send("CONNECT", retries=1)
+        except ClientConnectionError:
+            await self._term_receiver_task()
+            self.term()
+            raise
+
+    def send(self, message: str, retries: int | None = None) -> None:
+        self.loop.run_until_complete(self._send(message, retries))
+
+    async def process_message(self, msg: str) -> None:
+        pass
+
+    async def _receiver(self) -> None:
+        while True:
+            try:
+                _, raw_msg = await self.socket.recv_multipart()
+                if raw_msg == b"ACK":
+                    self._ack_event.set()
```

Comment:
We discussed whether this event and the receiver task was even neccessary. Since the socket handles both incoming events and "ACK" on outbound messages, it seems we need both. Maybe this can be refactored in the future. Very good work!